test/depayloader_pipeline_test.exs

```diff
@@ -30,30 +34,34 @@ defmodule Membrane.Element.RTP.H264.DepayloaderPipelineTest do
       glued_data = FUFactory.glued_fixtures()
       data_base = 1..10
 
-      data_base
-      |> Enum.flat_map(fn _ -> FUFactory.get_all_fixtures() end)
-      |> Enum.map(fn binary -> <<0::1, 2::2, 28::5>> <> binary end)
-      ~> (list -> Enum.zip(list, 1..Enum.count(list)))
-      |> Enum.map(fn {data, seq_num} -> into_rtp_buffer(data, seq_num) end)
-      |> generator_from_data()
-      |> DepayloaderTestingPipeline.start_pipeline()
-      ~> ({:ok, pid} -> Membrane.Pipeline.play(pid))
+      {:ok, pid} =
+        data_base
+        |> Enum.flat_map(fn _ -> FUFactory.get_all_fixtures() end)
+        |> Enum.map(fn binary -> <<0::1, 2::2, 28::5>> <> binary end)
+        ~> (list -> Enum.zip(list, 1..Enum.count(list)))
+        |> Enum.map(fn {data, seq_num} -> into_rtp_buffer(data, seq_num) end)
+        |> generator_from_data()
+        |> DepayloaderTestingPipeline.start_pipeline()
+
+      Membrane.Pipeline.play(pid)
 
       Enum.each(data_base, fn _ ->
-        assert_receive %Buffer{payload: data}
+        assert_sink_buffer(pid, :sink, %Buffer{payload: data})
         assert <<1::32, ^glued_data::binary()>> = data
       end)
     end
   end
 
   def generator_from_data(data) do
```

Comment:
Didn't get deep into this, but didn't we change source so that it accepted bare data also?

Reply:
Source expects either function or enum with `Membrane.Payload.t()`. Here I am providing buffers (metadata contains seqnums which are crucial for this element to work).